docs/howto/operator/custom-operator.rst

```diff
@@ -0,0 +1,185 @@
+ .. Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+ ..   http://www.apache.org/licenses/LICENSE-2.0
+
+ .. Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+
+Create Custom Operator
+=======================
+
+
+Airflow allows you to create new operators to suit the requirements of you or your team. 
+The extensibility is one of the many reasons which makes Apache Airflow powerful. 
+
+You can create any operator you want by extending the :class:`airflow.models.baseoperator.BaseOperator`
+
+There are two methods that you need to override in a derived class:
+
+* Constructor - Define the parameters required for the operator. You only need to specify the arguments specific to your operator.
+  Use ``@apply_defaults`` decorator function to fill unspecified arguments with ``default_args``. You can specify the ``default_args``
+  in the dag file. See :ref:`Default args <default-args>` for more details.
+
+* Execute - The code to execute when the runner calls the operator. The method contains the 
+  airflow context as a parameter that can be used to read config values.
+
+Let's implement an example ``HelloOperator``:
```

Comment:
The one thing we don't mention here is _where_ to define this class/which file to put this code in.

For simplicity it can be defined in the dag file, but we should mention/point at another doc of other places where it can be defined. See https://www.astronomer.io/guides/airflow-importing-custom-hooks-operators/